Automattic/Gravatar-SDK-Android
# OAuth token storage

Closes #235

### Description

In order to keep the user logged in we have to store the token that authenticates with the Gravatar API. 

To do that I've used DataStore preferences. The token is not encrypted for now. I will take a look at what's possible after this PR.

I've added another dependency - [Jetpack App Startup](https://developer.android.com/topic/libraries/app-startup). This one is used to initialize our library with the application context without forcing the user to init the library manually. 

I've also added the `GravatarQuickEditor.logout()` so third-party devs can clean the session for the given email. 


https://github.com/user-attachments/assets/2860b904-ff9d-4d8d-a175-57c81fbaa7cb



### Testing Steps

1. Start the demo app
2. Tap the "Update avatar" on the 3 tab
3. Follow the OAuth process (keep in mind there are two activities handling the redirect URL, make sure you pick the right one)
4. Confirm you see "Insert the real avatar picker here"
5. Tap "Done"
6. Tap the "Update avatar"
7. Confirm you see "Insert the real avatar picker here"
8. Tap "Done"
9. Tap "Logout user"
10. Tap the "Update avatar"
11. Confirm auth flow is started
